src/lib/libfock/yoshimine.cc

```diff
@@ -137,7 +142,79 @@ void Bucket::flush(const int lastbuf)
     in_bucket() = 0;
 }
 
+BucketLight::BucketLight() {
+    nints_ = 0;
+    ints_ = NULL;
+}
+
+BucketLight::~BucketLight() {
+    if(ints_) delete ints_;
+}
+
+void BucketLight::set_file(std::string name) {
+    std::stringstream ss;
+    ss <<  PSIOManager::shared_object()->get_default_path();
+    ss <<  "/";
+    ss << psi_file_prefix;
+    ss << ".";
+    ss << getpid();
+    ss << ".";
+    ss << PSIO::get_default_namespace();
+    ss << ".";
+    ss << "yosh.";
+    ss << name;
+    ss << ".dat";
+    filename_ = ss.str();
+    fh_ = fopen(filename_.c_str(),"wb");
+    fseek(fh_,0,SEEK_SET);
+}
+
+void BucketLight::close_file(int keep) {
+    fclose(fh_);
```

Comment:
I think this entire file should not even be there. The code history has been a bit chaotic.